code/modules/power/supermatter/supermatter_process.dm

```diff
@@ -53,14 +54,38 @@
 			env.merge(absorbed_gasmix)
 			air_update_turf(FALSE, FALSE)
 	else
+		combined_gas = absorbed_gasmix.total_moles()
+		gas_comp = list()
+
+		power_transmission_bonus = 0
+		dynamic_heat_modifier = 0
+		dynamic_heat_resistance = 0
+		gasmix_power_ratio = 0
+		powerloss_dynamic_scaling = 0
+		
+		for (var/gas_path in absorbed_gasmix.gases)
+			var/datum/sm_gas/sm_gas = GLOB.sm_gas_behavior[gas_path]
+			gas_comp[gas_path] = absorbed_gasmix.gases[gas_path][MOLES] / combined_gas
```

Comment:
I suggest another name for this variable: portion_of_total would work as well. gas_fraction; gas_share; gas_percentage; portion_of_total

Reply:
gas percentage sounds nice. Ill do it.